docs/src/main/asciidoc/hibernate-orm-panache.adoc

```diff
@@ -1,326 +1,752 @@
+    public static void deleteStefs(){
+        delete("name", "Stef");
+    }
+}
+----
+
+
+=== Using the Panache repository pattern
+Panache enables you to create entities that contain the model and repository in a single class. However, if you prefer to create a separate class for the model and a separate class for the repository and still have the other benefits of Panache, for example having the `getters` and `setter` automatically generate, you can use the repository pattern.
+
+.Procedure.
+
+
+. Add the Panache JPA extension dependency to your {ProductName} project `pom.xml` file:
++
+[source,java]
+----
+<dependencies>
+    <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-hibernate-orm-panache</artifactId>
+    </dependency>
+----
 
+. Define entities as JPA entities. The following example defines the `Person` class as a JPA entity.
++
+[source,java]
+----
+@Entity
+public class Person {
+    @Id @GeneratedValue private Long id;
+    private String name;
+    private LocalDate birth;
+    private Status status;
+
+    public Long getId(){
+        return id;
+    }
+    public void setId(Long id){
+        this.id = id;
+    }
+    public String getName() {
+        return name;
+    }
+    public void setName(String name) {
+        this.name = name;
+    }
+    public LocalDate getBirth() {
+        return birth;
+    }
+    public void setBirth(LocalDate birth) {
+        this.birth = birth;
+    }
+    public Status getStatus() {
+        return status;
+    }
+    public void setStatus(Status status) {
+        this.status = status;
+    }
+}
+----
+. Define your repository and add instance methods. The following example defines the `PersonRepository` :
++
+[source,java]
+----
+@ApplicationScoped
+public class PersonRepository implements PanacheRepository<Person> {
+
+   public Person findByName(String name){
+       return find("name", name).firstResult();
+   }
+
+   public List<Person> findAlive(){
+       return list("status", Status.Alive);
+   }
+
+   public void deleteStefs(){
+       delete("name", "Stef");
+  }
+}
+----
+
++
+. Add the following code to the REST endpoint where you want to use entity operations so that the operations that are defined in `PanacheEntityBase` are available:
++
+[source,java]
+----
+@Inject
+PersonRepository personRepository;
+
+@GET
+public long count(){
+    return personRepository.count();
+}
+----
+
+=== Pananche common operations
+This section lists common operations that you can use with your entity or repository. The active record pattern examples use a `Person` entity created from the `PanacheEntity` class. The repository pattern examples use a regular JPA `Person` entity and the `PersonRepository` created from the
+`PanacheRepository` class.
+
+* Create a record (active record pattern and repository pattern):
++
 [source,java]
 ----
-// creating a person
 Person person = new Person();
 person.name = "Stef";
```

Comment:
Actually for the Active-Record pattern you need to call the setters here.

Reply:
Oh. Can you tell me what to add for the active-record pattern. As you might have noticed, I combined the active-record and repository method lists. But in the source both have the same example for creating a person.

Reply:
You've split it now, which is fine. But the repository pattern usage samples need to call the setters now if the entity doesn't extend `PanacheEntity`. So `person.setName("Stef")` instead of `person.name = "Stef"`, and similarly for all fields.